src/config/ConfigManager.cpp

```diff
@@ -3,6 +3,9 @@
 #include <filesystem>
 #include <glob.h>
 #include <cstring>
+#if defined(_LIBCPP_VERSION)
```

Comment:
why is this guarded?

Reply:
I notice that it only required for libc++
Well my knowledge of c/c++ is a bit bad, that can be wrong. At least work on my side

Reply:
Well if yoi think is wrong I can take out

Reply:
Think about that I should use:
`#if defined(__clang__)` since on gcc it dosent produce a error without mutex

Reply:
just include it without guards

Reply:
Ok

Reply:
Done